agent/app/agent_compatibility_linux.go

```diff
@@ -61,3 +62,13 @@ func (agent *ecsAgent) checkCompatibility(engine engine.TaskEngine) error {
 
 	return nil
 }
+
+// scanPlugins returns the list of plugin names from docker pkg's plugin utilities method.
+func (agent *ecsAgent) scanPlugins() ([]string, error) {
+	pluginNames, err := plugins.Scan()
```

Comment:
can you make sure that `plugins.Scan` is behind an interface so that we can unit test that path as well? on that note, we need unit test coverage for this i think

Reply:
did you mean agent.scanPlugins to be behind an interface? plugins.Scan() is in docker's pkg.

Reply:
Capturing the offline conversation here for the sake of completeness. I meant defining an interface and having that implemented with `plugins.Scan()` in production code and mocking it out in test code. Example: https://github.com/aws/amazon-ecs-agent/blob/edc3e26016e363d7e8e6c044b11bde9bbb8853a8/agent/utils/oswrapper/os.go#L19

Reply:
Added interface, open to suggestions on better naming of it and its mock.